pg_search/src/index/directory/mvcc.rs

```diff
@@ -179,6 +181,16 @@ impl MVCCDirectory {
     pub(crate) fn all_entries(&self) -> HashMap<SegmentId, SegmentMetaEntry> {
         self.all_entries.lock().clone()
     }
+
+    /// Returns the [`AtomicUsize`] where the number of segments that survive [`load_metas()`]'
+    /// visibility checking gets stored once [`load_metas()`] has actually been called.
+    ///
+    /// An implementation detail behind the value calculation is that there's special casing for
+    /// [`MvccSatisfies::LargestSegment`] in that it will use the count of **all** "Snapshot"-visible
+    /// segments rather than `1` (one).
+    pub(crate) fn total_segment_count(&self) -> Arc<AtomicUsize> {
+        self.total_segment_count.clone()
```

Comment:
Nit: This provides mutable access to the field for consumers... it looks like that's not actually necessary, so maybe doing the `load` inline might be better constrained.